alphagov/pay-connector
# PP-13241 parity checker requires 3ds fields to match revision

## WHAT YOU DID
When the Historical Event Emitter is emitting an event for a charge in connector and the charge_events table shows either AUTHORISATION SUCCESS or AUTHORISATION REJECTED and the charge's requires_3ds value is false it should include a GATEWAY_DOES_NOT_REQUIRE_3DS_AUTHORISATION event.

How to test
Start pay local using this branch in connector.
Make new user, login and make a demo payment
Then make a POST request to the historical event emmiter using the following url http://localhost:9300/v1/tasks/historical-event-emitter?start_id=1&max_id=100&record_type=charge&do_not_retry_emit_until_duration=7200 and make sure you receive a 200 response.
Then if you check the logs in connector you should find "Gateway does not require 3DS Authorisation event emitted for [chargeExternalId=<your_charge_id]" in the logs.
